core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1993,18 +1993,15 @@ protected void convertFrom(
       return;
     }
 
-    // Skip over LATERAL conversion
-    if (from.getKind() == SqlKind.LATERAL) {
-      from = ((SqlCall) from).operand(0);
-    }
-
     final SqlCall call;
     final SqlNode[] operands;
     switch (from.getKind()) {
     case MATCH_RECOGNIZE:
       convertMatchRecognize(bb, (SqlCall) from);
       return;
-
+    case LATERAL:
+      convertFrom(bb, ((SqlCall) from).operand(0));
+      return;
```

Comment:
This is needed to work correctly with Coral's [HiveSqlToRelConverter](https://github.com/linkedin/coral/blob/c51d3b5b457e8e4af17d719cf85b79220c3aff12/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveSqlToRelConverter.java#L76) class which extends this class and overrides this function.